Big upgrade symfony 6.4

AlexMyddleware opened the pull request:
…onfig in .env because deprecated

AlexMyddleware committed:
no ticket: change the label between changing and generating rule in the workflow action edit and create

AlexMyddleware committed:
fix for changedata if edit action, now field add and remove and edit value work

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 remove swiftmailer config in .env because deprecated

AlexMyddleware committed:
update composer.json, lock, symfony lock, many oudtated dependencies

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 remove swiftmailer from bundles

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 remove swiftmailer from packages

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 further removing swiftmailer

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 update the security.yaml with the correct encoders and password hashers, also use app logout path

AlexMyddleware committed:
use request stack instead of session interface

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 update user command to use password hasher

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 create an app logout function even if unused for compatibility with new symfony

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use new encoder for security controller

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 add new implementations necessary for the user entity

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 update the routing in the kernel

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use new method for authentication

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use request stack for templates

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use request stack for bootstrap

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use new session myddleware with request

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 TRY, WILL NOT ACTUALLY PUSH modify managers and controllers to use new interfaces, but will not keep changes, just for testing purposes

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 change the acces controls to allow for login public access

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use attribute instead of annotation

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use attributes instead of annotation, remove logout in security controller

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 remove outdated supports function, allow for null response in authentication failure

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 update path for logout

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 fixed myd login form id in index

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use boolean instead of int for deleted, recommended by symfony because more efficient

Myddleware commented on src/Controller/ManagementSMTPController.php:
```diff
@@ -497,32 +499,38 @@ public function sendEmail($name, Swift_Mailer $mailer)
     protected function sendinblueSendMailByApiKey($textMail)
     {
-        // Get the email adresses of all ADMIN
-        $this->setEmailAddresses();
```
@AlexMyddleware , ici on privilégie l'envoie d'email en utilisant le token sendinblue, pas le paramètre SMTP.  On en rediscute quand on a le temps.

Myddleware commented on src/Manager/RuleManager.php:
```diff
@@ -1089,32 +1088,15 @@ public static function beforeSave($solutionManager, $data)
     // Mêmes paramètres en entrée que pour la fonction beforeSave sauf que l'on a ajouté les entrées ruleId et date de référence au tableau
     public static function afterSave($solutionManager, $data)
     {
-        // Contrôle sur la solution source
```
@AlexMyddleware , je vois que tu as enlevé de la logique. Il faudra qu'on en discute.

Myddleware commented on src/Manager/RuleManager.php:
```diff
@@ -1202,345 +1184,20 @@ public static function getBidirectionalRules($connection, $params, $solutionSour
      */
     protected function cancel($id_document)
-    {
-        $param['id_doc_myddleware'] = $id_document;
```
@AlexMyddleware , je vois beaucoup de suppression ici. On en reparle ;)

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use this entity manager as this->getDoctrine is deprecated in 6.4

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 more replace of get doctrine

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 bootstrap 5 instead of 4

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use request stack instead of get session in the service

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 doctrine replaced by entity manager in the flux controller

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 rework smpt part 1: use mailer instead of swiftmailer, and MAILER_DSN instead of MAILER_URL, save in .env.local is functionnal as well as test email

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 remove TRY: rule manager original state

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 UPDATE RULEMANAGER PART 1: use statement request stack

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 UPDATE RULEMANAGER PART 2: private request stack

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 UPDATE RULEMANAGER PART 3: request stack in consttruct

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 UPDATE RULEMANAGER PART 4 init request stack

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 UPDATE RULEMANAGER PART 5 in aftersave function, use request stack

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 UPDATE RULEMANAGER PART 6: use request stack in cancel function

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 UPDATE RULEMANAGER PART 5: function unlockdocument

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 UPDATE RULEMANAGER PART 8: changedeleteflag

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 UPDATE RULEMANAGER PART 9: runMyddlewareJob

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 UPDATE RULEMANAGER PART 10 : rerun

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 account using old version of the encoder, use new instead

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 user manager service used in the account was using old encoder, use new instead

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 update form for bootstrap 5, now need to fix function in rule by id

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 detailed config in twig for error below in rule creation

Alerte: An unexpected error has occured on the fields mapping : The template has no parent and no traits defining the "form_help" block. (C:\laragon\www\myddleware_NORMAL\vendor\symfony\twig-bridge\Resources\views\Form\bootstrap_5_layout.html.twig line 373)

AlexMyddleware committed:
Revert "https://myddleware.atlassian.net/browse/PROJ-637 detailed config in twig for error below in rule creation"

This reverts commit 3ef973be51404fab68504f73a677d8158dad3af5.

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 put bootstrap 5 first and form div layout second in an array in twig

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 get doctrine to entity manager

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 more get doctrine to entity manager

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 request stack injected in the aftersave upon rule creation

AlexMyddleware committed:
better param handling in ajax, and lots of console.logs

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use proper request stack for symfony 6.4

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 fix assign filter in the rule

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use the injected request stack intsead of solution manager

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use cleaner aproach to verify is request is present

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 connector controller use new syntax and get intermediate variable for clarity

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use new session to get the session in rerun function

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use session to create new session

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 revert to original rule keys that was changed in 14 mars 2025 cf description

commit: b72401574c26449fa4b51d5fea8a8ad1a7a4e194

Jobmanaguer line 979

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 add border for fields in connector

AlexMyddleware committed:
pale grey instead of black

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use textarea for description of workflow

AlexMyddleware committed:
lots of console.logs to handle the fields

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 rework form to properly handle the fields in the javascript, add json in both assets and public as a fallback

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 update style for both input and select just in case

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 text area for workflowaction description, add styling

AlexMyddleware committed:
new workflow action style

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 styling for the edit as well

AlexMyddleware committed:
use correct container for the div

AlexMyddleware committed:
area larger

AlexMyddleware committed:
change position of form rows, need to reorder form but only move parts, no additional styling needed

AlexMyddleware committed:
clean up libraries, remove old ones, upgrade recent ones, no need for jquery because already in yarn

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use brevo instead of sendinblue

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 in smpt controller, don't take te whole form use the data as an object

AlexMyddleware committed:
use new brevo client to send

AlexMyddleware committed:
make command fit symfony standards

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 add a finally to be able to close the job after using the command

AlexMyddleware committed:
missing curly brackend at the end of security controller, was caused by the rebase

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 update servicess to use the new mailer

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 update new controller to use brevo api key

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 update 2 factor to use new mailer instead of swiftmailer

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 brevo in service

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 remove session from twofactor

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use requets stack instead of session auth controller

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use request stack instead of session in listener

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 use brevo api key instead of sendinblue in tools managers

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 session get form requests stack local in two factor

AlexMyddleware committed:
’ 0146 in don't ask me again

AlexMyddleware committed:
https://myddleware.atlassian.net/browse/PROJ-637 add paragraph for device
